exasol_advanced_analytics_framework/event_handler/context/top_level_event_handler_context.py

```diff
@@ -0,0 +1,219 @@
+                 temporary_schema_name: str):
+        super().__init__(temporary_bucketfs_location,
+                         temporary_db_object_name_prefix,
+                         temporary_schema_name)
+
+    def _release_object(self, object_proxy: ObjectProxy):
+        self._check_if_valid()
+        self._valid_object_proxies.remove(object_proxy)
+        if object_proxy in self._owned_object_proxies:
+            self._owned_object_proxies.remove(object_proxy)
+        self._invalid_object_proxies.add(object_proxy)
+        object_proxy._invalidate()
+
+    def _register_object(self, object_proxy: ObjectProxy):
+        self._check_if_valid()
+        self._valid_object_proxies.add(object_proxy)
+
+    # TODO generate the cleanup query in the opposite order then they wete created
+    #  to make sure we remove first view which depend on other temporary things
+    def cleanup_released_object_proxies(self) -> List[Query]:
+        db_objects: List[DBObjectProxy] = \
+            [object_proxy for object_proxy in self._invalid_object_proxies
+             if isinstance(object_proxy, DBObjectProxy)]
+        bucketfs_objects: List[BucketFSLocationProxy] = \
+            [object_proxy for object_proxy in self._invalid_object_proxies
+             if isinstance(object_proxy, BucketFSLocationProxy)]
+        self._invalid_object_proxies = set()
+        self._remove_bucketfs_objects(bucketfs_objects)
+        return [object_proxy.get_cleanup_query() for object_proxy in db_objects]
+
+    def _remove_bucketfs_objects(self, bucketfs_object_proxies: List[BucketFSLocationProxy]):
+        for object_proxy in bucketfs_object_proxies:
+            object_proxy.cleanup()
+
+    def transfer_object_to(self, object_proxy: ObjectProxy,
+                           scope_event_handler_context: ScopeEventHandlerContext):
+        if self._is_child(scope_event_handler_context):
+            self._transfer_object_to(object_proxy, scope_event_handler_context)
+        else:
+            raise RuntimeError("Given ScopeEventHandlerContext not a child.")
+
+
+class _ChildEventHandlerContext(_ScopeEventHandlerContextBase):
+    def __init__(self, parent: ScopeEventHandlerContext,
+                 temporary_bucketfs_location: AbstractBucketFSLocation,
+                 temporary_db_object_name_prefix: str,
+                 temporary_schema_name: str):
+        super().__init__(temporary_bucketfs_location,
+                         temporary_db_object_name_prefix,
+                         temporary_schema_name)
+        self.__parent = parent
+
+    @property
+    def _parent(self) -> ScopeEventHandlerContext:
+        return self.__parent
+
+    def _release_object(self, object_proxy: ObjectProxy):
+        self._check_if_valid()
+        self._valid_object_proxies.remove(object_proxy)
+        self._invalid_object_proxies.add(object_proxy)
+        self._parent._release_object(object_proxy)
+
+    def _register_object(self, object_proxy: ObjectProxy):
```

Comment:
In case of implementing the register_object() in parent class, we should change here
```
super().register_object(object)
self._parent._register_object(object_proxy)
```

Reply:
so you mean we cal the extracted method from the super class and then we call self._parent_..., yeah, I think, this is a good idea, especially, if it should get more involved in the future.

Reply:
exactly